docs/API.md

```diff
@@ -1702,34 +1703,33 @@ The above example is equivalent to the `meta/common-info` example above.
 
 **Not yet implemented**
 
-#### POST *id*/resources/name.stream
+#### GET *id*/meta/resources
 
-Posting to the resources path creates a new version of the given stream
-for the charm with the given id. The request returns the new version.
+Getting from this path will return metadata regarding the resources for the charm with id of `id`.
 
 ```go
-type ResourcesRevision struct {
-        Revision int
+type ResourcesResponse struct {
+    Resources []Resource
 }
-```
 
-#### GET *id*/resources/name.stream[-revision]/arch/filename
+type Resource struct {
+    Name        string
+    Type        string
+    Path        string
+    Description string
+    Revision    int
+    Fingerprint string
+    Size        int64
+}
+```
 
-Getting from the `/resources` path retrieves a charm resource from the charm
-with the given id. If version is not specified, it retrieves the latest version
-of the resource. The SHA-256 hash of the data is specified in the HTTP response
-headers.
+#### GET *id*/resources/*name*[/revision]
 
-#### PUT *id*/resources/[~user/]series/name.stream-revision/arch?sha256=hash
+Getting from the `/resources` path retrieves a charm resource from the charm with the given id. If version is not specified, it retrieves the latest version of the resource. The SHA-384 hash of the data is specified in the HTTP response headers.
```

Comment:
Probably worth matching the /archive phrasing here:

    The `/resources` path returns the charm resource from the charm with the given id. etc.